app/js/filterFunctions.ts

```diff
@@ -117,7 +121,7 @@ function percentageFilter(resultArray: Array<any>) {
  * @return Array the filtered array of users
  */
 function testAllocatedFilter(resultArray: Array<any>) {
-    let filterTestAllocated = document.querySelector<HTMLInputElement>('#testAllocated')
+    let filterTestAllocated = document.querySelector<HTMLInputElement>('#testAllocated');
     let newResultArray = resultArray;
```

Comment:
This would be a good opportunity to use `reduce()` or `filter()`

Alternatively, if that is too hard to implement (fair!) - this can be cleaned up - you don't need to assign a variable to another variable and then later on down the line, re-assign it to an empty array... just return inside the if or return whatever you need to (an error? an empty array?) if the value of testAllocated isn't set

Reply:
changed to filter function